enhancements/kube-apiserver/stability-point-to-point-network-check.md

```diff
@@ -82,6 +82,80 @@ What are the caveats to the implementation? What are some important details that
 didn't come across above. Go in to as much detail as necessary here. This might
 be a good place to talk about core concepts and how they relate.
 
+#### RelatedEndpoint Custom Resource
+
+```go
+package v1alpha1
+import 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+// RelatedEndpoint references a "interesting" endpoint for a given pod.
+type RelatedEndpoint struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata"`
+    
+    Spec RelatedEndpointSpec `json:"spec"`
+    Status RelatedEndpointStatus `json:"status"`
+}
+
+type RelatedEndpointSpec struct {
+    Pod string `json:"string"`
+    Endpoint RelatedEndpointSpecEndpoint `json:"endpoint"`
+}
+
+type RelatedEndpointSpecEndpoint struct {
+    // URL is the url of the end point
+    URL string `json:"url"`
+    // CACertFile, if specified, will be used to validate server certs
+    CACertFile string `json:"url,omitempty"`
+    // CertFile, if specified, will be sent to the endpoint as a client cert
+    CertFile string `json:"url,omitempty"`
+    // KeyFile, if specified, will be used when CertFile is also specified
+    KeyFile string `json:"url,omitempty"`
+    // Reachable indicated that the endpoint was reachable
+    Reachable bool `json:"reachable"`
+}
+
+type RelatedEndpointStatus struct {
+    // Message provides status details
+    Message string `json:"message,omitempty"`
+    // Latency indicated the number of ms before the endpoint responded
+    Latency int `json:"latency"`
+    // LastTransitionTime indicated the last time the status was updated.
+    LastTransitionTime metav1.Time     `json:"lastTransitionTime,omitempty"`
+}
+
+```
+
+Example `RelatedEndpoint` CRs instance that an operator might create for one of its managed pods:
```

Comment:
You'll need to describe how this works for deployments and daemonsets.  I think the idea works, but you need to be specific.  I suspect you'll end up providing some kind of higher level logic that says, "all pods matching this label selector should have connectivity to these points"